src/cpp/lib/QtWidgets/QMenu/qmenu_wrap.cpp

```diff
@@ -67,3 +71,61 @@ Napi::Value QMenuWrap::addAction(const Napi::CallbackInfo& info) {
   // TODO: see if we need to return from here an pointer to qaction or not.
   return env.Null();
 }
+
+Napi::Value QMenuWrap::addActionWithName(const Napi::CallbackInfo& info) {
```

Comment:
Hi @gluaxspeed 
I dont think we should allow QAction creation with name here. 

The reason:
1. QAction inherits from QObject. 
2. In order for events to work with QObject we extend all classes that extend from QObject with our own custom class EventWidget.  
TLDR; you can take a look at naction, npushbutton, ncheckbox etc. Basically for events to work in js side we need to extend QAction to NAction and so on.

3. No when we create a QAction ourselves in JS side we actually create a NAction on c++ side. That is when you do new QAction() on js side, we do new NAction on C++ side. But when we do ```QMenu::addActionWithName  (this->instance->addAction(actionName.c_str()))) ``` 
Here we are telling Qt that here is a string actionName, please create a QAction for me. So Qt will create a QAction in C++ instead of NAction. This would cause issue with event handling in JS.

So the basic idea is that we never allow Qt to create a QAction automatically and instead we create an NAction always. I hope i didnt complicate this answer much, please do ask questions if I wasnt clear.


So my opnion is that we dont have addActionWithName as similar functionality can be achieved by creating a QAction first on the JS side and passing it via addAction. 

PS: I will push a commit and you will see what I mean by it.

Reply:
https://github.com/nodegui/nodegui/pull/347/commits/9e5e1da475d9f67a59d7a28b383fa4a9c580e265

Reply:
Ah okay, I believe I was only trying to do that because currently the menus don't show up in the right location and I thought it was because the actions weren't generated by Qt automatically. Instead, I will try a layout approach or some other way :)